src/tests/ftest/datamover/copy_procs.py

```diff
@@ -0,0 +1,163 @@
+    def create_cont(self, pool, path=None):
+        """Create a TestContainer object."""
+        # Get container params
+        container = TestContainer(
+            pool, daos_command=DaosCommand(self.bin))
+        container.get_params(self)
+
+        if path is not None:
+            container.path.update(path)
+        
+        # Create container
+        container.create()
+
+        # Save container
+        self.container.append(container)
+
+        return container
+
+    def test_copy_procs(self):
+        """
+        Test Description:
+            DAOS-TODO: Verify multi-process (rank) copying.
+        Use Cases:
+            Create pool.
+            Crate POSIX container1 and container2 in pool.
+            Create a single 100M file in container1 using ior.
+
+        :avocado: tags=all,pr,datamover
+        :avocado: tags=copy_procs
+        """
+        # Create pool and containers
+        pool1 = self.create_pool()
+        container1 = self.create_cont(pool1)
+        container2 = self.create_cont(pool1)
+
+        # Get the varying number of processes
+        procs_list = self.params.get("processes", "/run/datamover/copy_procs/*")
+
+        # Create the test file
+        self.write_daos(pool1, container1)
+        
+        # Run with varying number of processes
+        for num_procs in procs_list:
+            test_desc="copy_procs (processes={})".format(num_procs)
+            self.run_dcp(
+                src="/", dst ="/",
+                processes=num_procs,
+                src_pool=pool1, src_cont=container1,
+                dst_pool=pool1, dst_cont=container2,
+                test_desc=test_desc)
+            self.read_verify_daos(pool1, container2)
+
+    def write_daos(self, pool, container):
+        """Uses ior to write the test file to a DAOS container."""
+        self.ior_cmd.flags.update(self.flags_write)
+        self.ior_cmd.set_daos_params(self.server_group, pool, container.uuid)
+        out = self.run_ior(self.get_ior_job_manager_command(), self.processes)
+
+    def read_verify_daos(self, pool, container):
+        """Uses ior to read-verify the test file in a DAOS container."""
+        self.ior_cmd.flags.update(self.flags_read)
+        self.ior_cmd.set_daos_params(self.server_group, pool, container.uuid)
+        out = self.run_ior(self.get_ior_job_manager_command(), self.processes)
+
+    def run_dcp(self, src, dst,
+                processes,
+                src_pool=None, dst_pool=None, src_cont=None, dst_cont=None,
+                test_desc=None):
+        """Use mpirun to execute the dcp utility"""
+        # Set up the dcp command
+        dcp = DataMover(self.hostlist_clients)
+        dcp.get_params(self)
+        dcp.src_path.update(src)
+        dcp.dest_path.update(dst)
+        dcp.set_datamover_params(src_pool, dst_pool, src_cont, dst_cont)
+
+        # Run the dcp command
+        if test_desc is not None:
+            self.log.info("Running dcp: {}".format(test_desc))
```

Comment:
(pylint-logging-format-interpolation) Use % formatting in logging functions and pass the % parameters as arguments